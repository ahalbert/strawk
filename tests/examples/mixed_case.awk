BEGIN { sentence = 0}

/(?s)(.*)\./ { 
  sentence++
  s = sub(/\n/, "", $1)
  if sentence == 4 {
    new_sentence = ""
    for (char = 0; char < length(s); char++) {
      if char % 2  == 0 {
        c = toupper(substr(s, char, 1))
      } else {
        c = tolower(substr(s, char, 1))
      }
      new_sentence = new_sentence c
    }
    print new_sentence
  }
}
