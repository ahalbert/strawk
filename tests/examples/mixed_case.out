 mAxImE QuAsI MaGnI VeLiT EiUs nAm aUt eSsE VoLuPtAtIbUs qUiS VeLiT RePeLlEnDuS
